mlflow/sagemaker/__init__.py

```diff
@@ -1336,6 +1336,22 @@ def _get_sagemaker_config_name(endpoint_name):
     return f"{endpoint_name}-config-{get_unique_resource_id()}"
 
 
+def _get_sagemaker_config_tags(endpoint_name):
+    return [{"Key": "app_name", "Value": endpoint_name}]
+
+def _prepare_sagemaker_tags(
+  sagemaker_tags: Dict[str, str],
+  config_tags: List[Dict[str, str]]  
+):
+    """
+    Convert dict of tags to list for SageMaker resources and adds to config tags list.
+    """
+    if sagemaker_tags:
+        sagemaker_tags = [{"Key": key, "Value": str(value)} for key, value in sagemaker_tags.items()]
+        config_tags.extend(sagemaker_tags)
```

Comment:
What does this line do? 
The return fo the private function is `sagemaker_tags`, which is handled entirely by lines 1349 and 1350. 
Was the intention to return config_tags ?

Reply:
This adds any custom tags that are passed in the initial create_deployment function (example [here](https://github.com/mlflow/mlflow/blob/f3fc6aa1983baf24e84a1477b0920226f977b46b/tests/sagemaker/test_sagemaker_deployment_client.py#L436-#L442) ) to also be added to the SageMaker configuration tags. With the current functionality, the only tags being added to the SageMaker config are manually set [here](https://github.com/mlflow/mlflow/blob/f3fc6aa1983baf24e84a1477b0920226f977b46b/mlflow/sagemaker/__init__.py#L1697)

Reply:
```python
_prepare_sagemaker_tags({"a_config_tag": "a_value", "my_custom_tag": "my_custom_value"}, [{"Key": "app_name", "Value": "my_app"}])
```

```shell
[{'Key': 'a_config_tag', 'Value': 'a_value'},
 {'Key': 'my_custom_tag', 'Value': 'my_custom_value'}]
```

If this is the intended output, then there is no need to have ``config_tags`` (which is the structure ``[{"Key": "app_name", "Value": <endpoint_name>}]`` defined and returned within ``_get_sagemaker_config_tags``)